### DISCUSSION TITLE
Changes relating to how one defines a Monoid

### DISCUSSION THREAD

**Discussion by chrisdenman** [learnyouahaskell/learnyouahaskell.github.io#72] [Ideas]
I'm keen on updating the text to reflect the fact that since March 2018 `GHC 8.4.1 (base-4.11.0.0)` - `Semigroup` is a superclass of `Monoid` which means that to make a type an instance of `Monoid`, it also has to be a `Semigroup`.

It's not a major issue IMHO. When I worked through LYAH (in concert with some exercises) it wasn't too onerous to complete the puzzle. However, this may not be true for everyone else.

Chapter 11 is quite tricky, it introduces a lot of new concepts. I feel that even minor stumbling blocks at this point may deter some from completion; after having completed the bulk of the book, this would be a shame.

I haven't reviewed the relevant portions of the text recently... but if there is a consensus that this change would be worthwhile, I'd happily have a first-crack at it.

I gratefully welcome any input.

**Comment by memowe** [comment]
As this is a rather old breaking change I would absolutely vote for that change. Very much looking forward to your take. :)

**Reply by chrisdenman** [reply]
Thanks. i'll get going on it.
